docs/recipes.md

```diff
@@ -589,4 +589,31 @@ it has not seen, it will compute the predictions and store them in a cache. The
 next time it is called with the same parameters, instead of recomputing the
 predictions, it will return the previously computed value from the cache.
 
+Unlike `functools.cache`, `mo.cache` is notebook aware; and will become
+invalidated if a relevant global variable is changed (including State and UI
+Elements), or the function definition is changed.
+
 See our [best practices guide](guides/best_practices) to learn more.
+
+### Persistent caching for very expensivve computations
+
+**Use case.** If you are using marimo to capture very compute intensive
+results, you may want to save the state of your computations to disk. Ideally,
+if you update your code, then this save should be invalidated. It may also be
+advantageous to add UI elements to explore your results, without having to
+recompute expensive computations.
```

Comment:
Where does the cache get stored on disk? So users can clear it if needed.

We have a `.marimo` directory now, in which HTML snapshots get saved. Maybe it should be under there, if it isn't already?

Reply:
Good place to save, just in `.outputs` right now

Reply:
Actually playing with this, maybe it's good for the cache to not be in a hidden directory? Ie maybe something like your default is fine. Since some of the objects you'll be caching will be rather large. Or, maybe we take a page out of Python's book and choose `__mocache__` or `__marimo__`..

Reply:
Should the directory structure be

- `__marimo__` (or whatever it's called)
  - `<notebook_name>`
    - `<cache_name>`
    
? That is should we group caches belong to the same notebook in an intermediate directory?

Reply:
Yeah, there's not a great mechanism for clearing out the cache, so a hidden folder might not be the move. I was thinking of putting in an LRU, but diskspace is relatively cheap now. However, there are still cases where you just want to clean up. No folder potentially needed for remote caching (or maybe optional/limited local LRU)

In the "execution mode" (or native save?) extension to this, I also save code content. If the graph is saved too, then this could be leveraged into a sophisticated check-pointing mechanism/audit (restore your notebook, output, state, and all the variables). Jupyter's `.ipynb_checkpoints` folder makes it clear that it's used for checkpoints, so I worry that `__mocache__` could be misleading if caching expands, and `__marimo__` is maybe not descriptive enough.

Haven't gotten back to review edits (maybe this evening), so plenty of time to think about this. I think all of them work, but once decided, ideally shouldn't be changed for backwards compat.

Reply:
How's

 - .marimo/
   - cache/
       - `<cache name>` / `<notebook name>`

Reply:
That sounds good to me.

Did you land on storing in the hidden `.marimo`? I'll leave it up to you; whatever you think is best.